src/v_video.c

```diff
@@ -63,6 +64,12 @@ int dirtybox[4];
 // This is needed for Chocolate Strife, which clips patches to the screen.
 static vpatchclipfunc_t patchclip_callback = NULL;
 
+// We use a different scaler for PNG screenshots from screen rendering.
+// Default value is the smallest required for a lossless aspect-correct shot.
+static screen_mode_t * screenshot_scaler;
```

Comment:
Thanks for the review; I've fixed that and two more like it, rebased onto current master and force pushed (the fixes are squashed into the latest commit of the branch)